lib/main/atom/views/outline/navigationTreeComponent.tsx

```diff
@@ -0,0 +1,409 @@
+  private activeEditorChanging: Disposable
+
+  constructor(public props: Props) {
+    this.selectedNode = null
+    this.prepareNavTree(props.navTree)
+    etch.initialize(this)
+    atom.workspace.observeActiveTextEditor(this.subscribeToEditor)
+  }
+
+  public async update(props: Partial<Props>) {
+    if (props.navTree) {
+      this.prepareNavTree(props.navTree)
+    }
+    this.props = {...this.props, ...props}
+    await etch.update(this)
+  }
+
+  public async destroy() {
+    if (this.editorScrolling) {
+      this.editorScrolling.dispose()
+    }
+    if (this.editorChanging) {
+      this.editorChanging.dispose()
+    }
+    if (this.activeEditorChanging) {
+      this.activeEditorChanging.dispose()
+    }
+    this.selectedNode = null
+    await etch.destroy(this)
+  }
+
+  private setEditor(editor: TextEditor) {
+    this.editor = editor
+  }
+
+  private async setNavTree(navTree: NavigationTreeViewModel | null) {
+    this.prepareNavTree(navTree)
+    if (isEqual(navTree, this.props.navTree)) {
+      return
+    }
+    this.restoreCollapsed(navTree, this.props.navTree)
+    this.props.navTree = navTree
+    this.selectedNode = null
+    await etch.update(this)
+  }
+
+  public async forceUpdate() {
+    await etch.update(this)
+  }
+
+  private async loadNavTree(filePath?: string) {
+    filePath = filePath ? filePath : this.editor.getPath()
+    if (filePath) {
+      try {
+        const client = await clientResolver.get(filePath)
+        await client.executeOpen({file: filePath})
+        const navtreeResult = await client.executeNavTree({file: filePath as string})
+        const navTree = navtreeResult ? navtreeResult.body! : undefined
+        if (navTree) {
+          this.setNavTree(navTree as NavigationTreeViewModel)
+        }
+      } catch (err) {
+        console.error(err, filePath)
+      }
+    }
+  }
+
+  /**
+   * HELPER modify / prepare NavigationTree for rendering.
+   *
+   * E.g. sort childItems by their location, preprocess className-string
+   *
+   * @param {NavigationTreeViewModel} navTree
+   *            the NavigationTree that will be prepared for rendering
+   */
+  private prepareNavTree(navTree: NavigationTreeViewModel | null): void {
+    if (navTree === null) return
+    navTree.styleClasses = this.getIconForKind(navTree.kind)
+    const modifiersClasses = this.getClassForKindModifiers(navTree.kindModifiers)
+    if (modifiersClasses) {
+      navTree.styleClasses += " " + modifiersClasses
+    }
+
+    if (navTree.childItems) {
+      if (navTree.childItems.length < 1) {
+        // normalize: remove empty lists
+        navTree.childItems = undefined
+        return
+      }
+
+      // TODO should there be a different sort-order?
+      //     for now: sort ascending by line-number
+      navTree.childItems.sort((a, b) => this.getNodeStartLine(a) - this.getNodeStartLine(b))
+
+      let child: NavigationTreeViewModel
+      for (child of navTree.childItems) {
```

Comment:
will change 

(was just a matter of taste, as it feels weird using `const` in these cases -- it seems inconsistent to me, since the variable-reference is _not_ constant, and will get re-set in each loop)

Reply:
The way I see it, `const` is an intent specifier. I.e. "`const`" means "I am not going to reassign this variable". I will concede that it's very javascript-y, and not at all consistent with other languages. But, personally, given a choice between code that makes sense in my head and code that is idiomatic, I always try to go for the latter, because more likely than not, I'm not the only person who's going to be hacking on that code, and idioms make life easier.